pkg/util/header_encoder.go

```diff
@@ -0,0 +1,24 @@
+package util
```

Comment:
This feels like a lot wrapping around  a simple `strings.Join(values, h.arraySeparator)`.  If we move to only doing this at the last moment before setting a header, is this best as a private helper in oauth2proxy.go that manages settings headers from a session?

Reply:
I agree I did not like it either :sob: but I like the idea of doing it later, that will greatly clean things up! let me get to that tonight.